### PR TITLE
Fix race in creating calls

### PR DESCRIPTION
**Pull request by dbkr**
Second attempt, just put a mutex around the method that processes member state events. The previous change meant that uninitialised calls were sitting in the map which is not a thing this code is prepared to handle and was causing test failures.

Also more logging.

Supersedes https://github.com/matrix-org/matrix-js-sdk/pull/2662

## Checklist

* [ ] Tests written for new code (and old code if feasible)
* [ ] Linter and other CI checks pass
* [ ] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-js-sdk/blob/develop/CONTRIBUTING.md))

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Fix race in creating calls ([\#2663](https://github.com/matrix-org/matrix-js-sdk/pull/2663)).

### CODE REVIEW

File: src/webrtc/groupCall.ts [hunk 1]
```diff
@@ -185,6 +185,9 @@ export class GroupCall extends TypedEventEmitter<
     private resendMemberStateTimer: ReturnType<typeof setInterval> | null = null;
     private initWithAudioMuted = false;
     private initWithVideoMuted = false;
+    // We use this as a set of mutexes, effectively, to make sure we only enter the function
+    // to process a member state event once at any one time for each member
+    private processingMemberStateEvent = new Map<string, boolean>();
```
Comment (SimonBrandner) [inline src/webrtc/groupCall.ts:190]: Why is this a map? Wouldn't a `Set` work?

File: src/webrtc/groupCall.ts [hunk 2]
```diff
@@ -754,6 +759,25 @@ export class GroupCall extends TypedEventEmitter<
...
+            logger.debug(`Already processing meber state event for ${event.getStateKey()}`);
+            return;
+        }
+
+        this.processingMemberStateEvent.set(event.getStateKey(), true);
+        try {
+            return this.processMemberStateEventInternal(event);
+        } finally {
+            this.processingMemberStateEvent.delete(event.getStateKey());
+        }
```
Comment (SimonBrandner) [inline src/webrtc/groupCall.ts:777]: What's the benefit of try/finally here?

**Review by SimonBrandner — COMMENTED** [review]
Otherwise, LGTM

**Comment by SimonBrandner** [comment]
Also, what if the event changes while we're processing it?

**Comment by dbkr** [comment]
Hmm, yes, you're right. We need to handle the newer ones somehow.

**Comment by dbkr** [comment]
Going back to https://github.com/matrix-org/matrix-js-sdk/pull/2662 again

**Pull request closed without merging**